### PR TITLE
Add System Information to sysperf command

### PR DESCRIPTION
**Pull request by 7suyash7**
Add System Information to the sysperf command
#299 

## ✅ I have completed the following steps:

* [✅] Run `make lint`
* [✅] Run `make test`
* [ ] Added tests (if applicable)

### CODE REVIEW

File: crates/rbuilder/src/live_builder/mod.rs [hunk 1]
```diff
@@ -202,9 +202,9 @@ where
                 continue;
             }
             // Allow only increasing blocks
-            if last_processed_block.map_or(false, |last_processed_block| {
```
Comment (ferranbt) [inline crates/rbuilder/src/live_builder/mod.rs:205]: Is this change required?

File: crates/rbuilder/src/utils/error_storage.rs [hunk 2]
```diff
@@ -83,10 +83,10 @@ pub fn store_error_event<T: serde::Serialize>(category: &str, error: &str, paylo
                 return;
             }
         };
-        if payload_json.as_bytes().len() > MAX_PAYLOAD_SIZE_BYTES {
```
Comment (ferranbt) [inline crates/rbuilder/src/utils/error_storage.rs:86]: Same as before, is this change required?

File: crates/sysperf/src/lib.rs [hunk 3]
```diff
@@ -39,6 +42,16 @@ pub struct CpuBenchmarkResult {
     pub threads_used: usize,
 }
 
+#[derive(Debug)]
+pub struct SystemInfoResult {
```
Comment (ferranbt) [inline crates/sysperf/src/lib.rs:46]: Could you also return disks and network interfaces?

File: crates/rbuilder/src/live_builder/mod.rs (hunk 1 above)
Comment (7suyash7) [reply crates/rbuilder/src/live_builder/mod.rs:205]: Yeah, `make lint` kept failing if I didn't address this... Probably due to Clippy (with -D warnings) treating warnings as errors

File: crates/rbuilder/src/utils/error_storage.rs (hunk 2 above)
Comment (7suyash7) [reply crates/rbuilder/src/utils/error_storage.rs:86]: `make lint` error again

File: crates/sysperf/src/lib.rs (hunk 3 above)
Comment (7suyash7) [reply crates/sysperf/src/lib.rs:46]: Yeah, sure!

**Review by ferranbt — APPROVED** [review]
